pkg/oc/bootstrap/docker/up.go

```diff
@@ -358,81 +357,34 @@ func (c *CommonStartConfig) Complete(f *osclientcmd.Factory, cmd *cobra.Command,
 	// Get a Docker client.
 	// If a Docker machine was specified, make sure that the machine is running.
 	// Otherwise, use environment variables.
-	taskPrinter.StartTask("Create Docker client")
+	taskPrinter.StartTask("Getting a Docker client")
 	client, err := getDockerClient(out, c.DockerMachine, true)
 	if err != nil {
 		return taskPrinter.ToError(err)
 	}
 	c.dockerClient = client
 	taskPrinter.Success()
 
-	// TODO HOLY SIDE-EFFECTS this is setting global bool values as part of a check!
-	// Check that we have the minimum Docker version available to run OpenShift
-	taskPrinter.StartTask("Checking Docker version")
-	if err := c.CheckDockerVersion(out); err != nil {
-		return taskPrinter.ToError(err)
-	}
-	taskPrinter.Success()
-
-	taskPrinter.StartTask("Checking OpenShift client")
-	if err := checkOpenShiftClient(out); err != nil {
-		return taskPrinter.ToError(err)
-	}
-	taskPrinter.Success()
-
-	if c.PortForwarding {
-		taskPrinter.StartTask("Checking prerequisites for port forwarding")
-		if err := checkPortForwardingPrerequisites(out); err != nil {
-			return taskPrinter.ToError(err)
-		}
-		taskPrinter.Success()
-	}
-
-	// Ensure that the OpenShift Docker image is available.
-	// If not present, pull it.
-	taskPrinter.StartTask(fmt.Sprintf("Checking for %s image", c.openshiftImage()))
-	if err := c.CheckOpenShiftImage(out); err != nil {
-		return taskPrinter.ToError(err)
-	}
-	taskPrinter.Success()
-
-	// Ensure that the Docker daemon has the right --insecure-registry argument.
-	// If not, then exit.
-	if !c.SkipRegistryCheck {
-		taskPrinter.StartTask("Checking Docker daemon configuration")
-		if err := c.CheckDockerInsecureRegistry(out); err != nil {
+	// Check whether the Docker host has the right binaries to use Kubernetes' nsenter mounter
+	// If not, use a shared volume to mount volumes on OpenShift
+	if isRedHatDocker, err := c.DockerHelper().IsRedHat(); err == nil && isRedHatDocker {
```

Comment:
What's the point of this check, before checking docker version. If you don't pass docker check, this command might fail as well giving you wrong error.

Reply:
Version check is in Check method (it does not mutate the struct).... i need to check if we need c.UseNsenterMount elsewhere

Reply:
(we do, so this check is correct)